src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/RedirectHandler.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http
+{
+    internal sealed partial class RedirectHandler : HttpMessageHandler
+    {
+        private readonly HttpMessageHandler _initialInnerHandler;       // Used for initial request
+        private readonly HttpMessageHandler _redirectInnerHandler;      // Used for redirects; this allows disabling auth
+        private readonly int _maxAutomaticRedirections;
+
+        public RedirectHandler(int maxAutomaticRedirections, HttpMessageHandler initialInnerHandler, HttpMessageHandler redirectInnerHandler)
+        {
+            Debug.Assert(initialInnerHandler != null);
+            Debug.Assert(redirectInnerHandler != null);
+
+            if (maxAutomaticRedirections < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(maxAutomaticRedirections));
+            }
```

Comment:
Can this just be an assert (for >= 1)?  In what situation would a negative maxAutomaticRedirections make it into here?  It's validated at https://github.com/dotnet/corefx/blob/cfe3ac1dd76cebe61a2206c0549085a669770a3b/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs#L130-L133 to be >= 1.

Reply:
Yep